File: ui/sustainability-ui/src/app/models/indicator.ts
export class Indicator {
  id: string;
  name: string;
  comment: string;
  unit: string;
  topicId: string;

  constructor(id: string, name: string, comment: string, unit: string, topicId: string) {
    this.id = id;
    this.name = name;
    this.comment = comment;
    this.unit = unit;
    this.topicId = topicId;
  }
}

File: java/src/main/java/ch/planval/datavisualizer/model/Indicator.java
package ch.planval.datavisualizer.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;
import lombok.ToString;
import org.hibernate.annotations.GenericGenerator;

import java.util.UUID;

@Getter
@Setter
@ToString
@Table(name="INDICATORS")
@Entity()
@AllArgsConstructor
@RequiredArgsConstructor
@Builder(toBuilder = true, builderClassName = "Builder")
public class Indicator {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;
    private String name;
    private String comment;
    private String unit;
    private UUID topicId;
}
